tests/commands/test_edit.py

```diff
@@ -0,0 +1,225 @@
+import os
+import subprocess
+import sys
+from pathlib import Path
+from onyo import commands  # noqa: F401
+
+from onyo.lib import Repo, OnyoInvalidRepoError
+
+onyo_root = "./"
+
+
+def test_edit_single_asset(helpers):
+    """
+    Check that for different paths it is possible to call `onyo edit` on a
+    single asset file.
+    """
+    dirs = ['simple',
+            's p a c e s',
+            's p a/c e s',
+            'very/very/very/deep/'
+            ]
+
+    files = ['laptop_apple_macbookpro.0',
+             'simple/laptop_apple_macbookpro.1',
+             's p a c e s/laptop_apple_macbookpro.2',
+             's p a/c e s/laptop_apple_macbookpro.3',
+             'very/very/very/deep/laptop_apple_macbookpro.4'
+             ]
+
+    helpers.populate_repo(onyo_root, dirs, files)
+    os.environ['EDITOR'] = "printf 'key: single_asset' >"
+
+    try:
+        repo = Repo(onyo_root)
+    except OnyoInvalidRepoError:
+        sys.exit(1)
+    assert len(repo.assets) > 0
+
+    # check edit for a single asset
+    for asset in repo.assets:
+        ret = subprocess.run(['onyo', 'edit', asset], input='y', capture_output=True, text=True)
+        assert ret.returncode == 0
+        assert "Save changes?" in ret.stdout
+        assert "+key: single_asset" in ret.stdout
+        assert not ret.stderr
+
+        # verify the changes were actually written:
+        ret = subprocess.run(['cat', asset], capture_output=True, text=True)
+        assert 'key: single_asset' in ret.stdout
+
+    # verify onyo repo is in a clean state
+    assert not repo.fsck()
+
+
+def test_edit_multiple_assets():
+    """
+    Check that it is possible to call `onyo edit` with a list of multiple assets
+    containing different file names at once.
+    """
+    os.environ['EDITOR'] = "printf 'key: multiple_assets' >"
+    try:
+        repo = Repo(onyo_root)
+    except OnyoInvalidRepoError:
+        sys.exit(1)
+
+    # check edit for a list of assets all at once
+    ret = subprocess.run(['onyo', 'edit'] + list(repo.assets), input='y', capture_output=True, text=True)
+    assert ret.returncode == 0
+    assert "Save changes?" in ret.stdout
+    assert "+key: multiple_assets" in ret.stdout
```

Comment:
This string should be present `n` times, correct? If so, a `count` would be more appropriate here.